src/environment.cc

```diff
@@ -105,6 +105,17 @@ std::string Environment::GetMetadataString(const std::string& path) const {
   }
 }
 
+bool Environment::IsGcpLocationZonal(const std::string& location) const {
+  // Use the number of dashes to distinguish between zones (e.g.
+  // "us-central1-a", "us-east1-b") and regions (e.g."us-central1", "us-east1").
+  int num_dashes = std::count(location.begin(), location.end(), '-');
+  if (num_dashes > 2) {
+    LOG(WARNING) << "Found more than two dashes in the GCP location: "
```

Comment:
Could even be `ERROR` — your basic assumptions are violated.

Reply:
Done